packages/store/src/synced.ts

```diff
@@ -1,11 +1,37 @@
 import {writable} from "svelte/store"
 import {getJson, setJson} from "@welshman/lib"
 
-export const synced = <T>(key: string, defaultValue: T) => {
-  const init = getJson(key)
-  const store = writable<T>(init === undefined ? defaultValue : init)
+export interface StorageProvider {
+  get: (key: string) => Promise<any>
+  set: (key: string, value: any) => Promise<void>
+}
+
+export interface SyncedConfig {
+  key: string
+  storage: StorageProvider
+  defaultValue: any
+}
```

Comment:
Let's use generics here:
```suggestion
export interface SyncedConfig<T> {
  key: string
  storage: StorageProvider
  defaultValue: T
}
```